src/poker.js

```diff
@@ -1,5 +1,116 @@
+      if (!handIsPair) return;
+
+      if (handIsTrio && handCardValue > currentWinningScoreTrio) {
+        currentWinningScoreTrio = handCardValue;
+        currentWinningHandTrio = hand;
+      }
+
+      if (handIsPair && handCardValue > currentWinningScorePair) {
+        currentWinningScorePair = handCardValue;
+        currentWinningHandPair = hand;
+      }
+    });
+
+    if (currentWinningScorePair === 0) return [];
+
+    if (currentWinningScorePair > 0 && currentWinningScoreTrio > 0)
+      return currentWinningHandTrio;
+
+    if (currentWinningScorePair > 0 && currentWinningScoreTrio === 0)
+      return currentWinningHandPair;
+  }
+  // -----------------------------------------------------------------------------------------------------------
+
+  /** ----------------------------------------------------------------------------------------------------------
+   * checks if the card hand (a single array with 2 strings denoting 'cards') is a pair or not
+   * @param {array} arr takes in a card hand array
+   * @returns boolean
+   */
+  isPair(arr) {
+    const [card1, card2] = arr;
+
+    return card1 === card2;
+  }
+  // -------------------------------------------------------------------------------------------------------------
+
+  /** ------------------------------------------------------------------------------------------------------------
+   * checks if the card hand (a single array with 3 strings denoting 'cards') is a trio or not
+   * @param {array} arr takes in a card hand array
+   * @returns boolean
+   */
+  isTrio(arr) {
+    const [card1, card2, card3] = arr;
+
+    return card1 === card2 && card2 === card3;
+  }
+  //  ------------------------------------------------------------------------------------------------------------
+
+  /** ------------------------------------------------------------------------------------------------------------
+   * returns the card value as an integer for the card hand array passed into it (ONLY USED WHEN YOU KNOW IT'S A PAIR)
+   * @param {array} arr takes in a card hand
+   * @returns integer
+   */
+  cardValue(arr) {
+    const [card1, card2] = arr;
+
+    if (card1 === card2) return cardPointMap[card1];
+
+    return false;
+  }
+  // ------------------------------------------------------------------------------------------------------------
 }
 
-module.exports = Poker
+// winningPair(hand1, hand2) {
+//   const handOneIsPair = this.isPair(hand1);
+//   const handTwoIsPair = this.isPair(hand2);
+//   const handOneCardValue = this.cardValue(hand1);
+//   const handTwoCardValue = this.cardValue(hand2);
+
+//   if (!handOneIsPair && !handTwoIsPair) return [];
+
+//   if (handOneIsPair && handTwoIsPair) {
+//     if (handOneCardValue > handTwoCardValue) return hand1;
+//     if (handOneCardValue < handTwoCardValue) return hand2;
+//   }
+
+//   if (handOneIsPair || handTwoIsPair) {
+//     if (handOneIsPair) return hand1;
+//     if (handTwoIsPair) return hand2;
+//   }
+// }
+
```

Comment:
Note for the whole code:
My first impression was that its very long so I got scared, but then I noticed it actually just the comments that makes it look long. As for the content itself, I cannot suggest anything to change.